commons/statistics/ols.go

```diff
@@ -0,0 +1,28 @@
+package statistics
+
+type Point interface {
+	X() float64
+	Y() float64
+}
+
+func LeastSquares(series []Point) (m float64, b float64) {
```

Comment:
I suggest this function take a "Points" interface, rather than a slice of Points.  The X and Y functions would take an index as a parameter.